frontend/src/component/feature/FeatureToggleList/FeatureToggleFilters/AddFilterButton/AddFilterButton.tsx

```diff
@@ -2,22 +2,26 @@ import React, { useState } from 'react';
 import Button from '@mui/material/Button';
 import Menu from '@mui/material/Menu';
 import MenuItem from '@mui/material/MenuItem';
-import { IFilterVisibility, IFilterItem } from '../FeatureToggleFilters';
-import { Box, styled } from '@mui/material';
+import { styled } from '@mui/material';
 import { Add } from '@mui/icons-material';
 
 const StyledButton = styled(Button)(({ theme }) => ({
     margin: theme.spacing(-1, 0, -1, 0),
     padding: theme.spacing(1.25),
 }));
+
 interface IAddFilterButtonProps {
```

Comment:
this state model seems to allow impossible states. We don't prevent at the type level a scenario where something is both on the visible and hidden list.

Reply:
Background: Great presentation on making impossible states impossible "https://www.youtube.com/watch?v=IcgmSRJHu_8&ab_channel=elm-conf"

Reply:
I feel like this is more runtime issue not compile time issue. Other than being super cool, I do not see how this will benefit or how to implement it even. I could play around with Exclude<>, but this is not for react states.